libs/shared/ui/src/lib/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -43,23 +55,43 @@ const DropdownMenuItem = forwardRef<ElementRef<typeof DropdownMenuPrimitive.Item
   }
 )
 
+interface DropdownMenuItemIconProps
+  extends VariantProps<typeof dropdownMenuItemVariants>,
+    Omit<ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Item>, 'color'> {
+  icon: ReactElement
+}
+
+const DropdownMenuItemIcon = forwardRef<ElementRef<typeof DropdownMenuPrimitive.Item>, DropdownMenuItemIconProps>(
```

Comment:
I think we can merge this into `DropdownMenuItem` because if `icon` becomes optional there is no need for 2 components